lisp/ess-r-mode.el

```diff
@@ -476,6 +481,12 @@ will be prompted to enter arguments interactively."
    (format
     "\n(R): ess-dialect=%s, buf=%s, start-arg=%s\n current-prefix-arg=%s\n"
     ess-dialect (current-buffer) start-args current-prefix-arg))
+  (unless (executable-find inferior-ess-r-program)
```

Comment:
This setq side effect here is not good. Such an important variable should be set in one place and not interfered with.

Reply:
I think this logic can be used whenever needed without setting `inefrior-ess-r-program`.

Reply:
Agreed in principal, but I'm not sure the best way to handle it. One option is to put something like this in the defcustom, which will increase the time it takes to load ESS.

> this logic can be used whenever needed without setting inefrior-ess-r-program

probably, but it sounds like more work than I'm willing to put in for an evil, proprietary operating system. :-)